pkg/resources/task_state_upgraders.go

```diff
@@ -0,0 +1,49 @@
+package resources
+
+import (
+	"context"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+)
+
+func v098TaskStateUpgrader(ctx context.Context, rawState map[string]any, meta any) (map[string]any, error) {
+	if rawState == nil {
+		return rawState, nil
+	}
+
+	rawState["condition"] = rawState["when"]
+	rawState["started"] = booleanStringFromBool(rawState["enabled"].(bool))
+	rawState["allow_overlapping_execution"] = booleanStringFromBool(rawState["allow_overlapping_execution"].(bool))
+	if rawState["after"] != nil {
+		newAfter := make([]string, len(rawState["after"].([]any)))
+		for i, name := range rawState["after"].([]any) {
```

Comment:
where these before required to be fully qualified in the config or not? If so, we need some migration guide informing about the config changes

Reply:
Good point, added.